pkg/archive/changes_unix.go

```diff
@@ -25,3 +26,21 @@ func statDifferent(oldStat *system.StatT, newStat *system.StatT) bool {
 func (info *FileInfo) isDir() bool {
 	return info.parent == nil || info.stat.Mode()&syscall.S_IFDIR != 0
 }
+
+// GetInode returns inode number from os.FileInfo object
+func GetInode(fi os.FileInfo) (inode uint64) {
+	s, _ := fi.Sys().(*syscall.Stat_t)
+	inode = uint64(s.Ino)
+	return
+}
+
+// IsHardlink detects if os.FileInfo object is hardlink or not
+func IsHardlink(fi os.FileInfo) bool {
+	switch sys := fi.Sys().(type) {
+	case *syscall.Stat_t:
+		if fi.Mode().IsRegular() && sys.Nlink > 1 {
```

Comment:
But directory can't have `Nlink > 0`. But unix sockets can, so I think it's excluding real hardlinks.